node/repo/fsrepo.go

```diff
@@ -172,6 +242,27 @@ func (fsr *FsRepo) initKeystore() error {
 	return os.Mkdir(kstorePath, 0700)
 }
 
+func (fsr *FsRepo) initVersion(t RepoType) error {
```

Comment:
Since most repos don't have version, we should be calling this on open too, which should gradually get users the version file in their repos (and eventually after next network upgrade or two we should be able to drop that)

Reply:
The closest thing to a repo `Open()` method is `Exists()`, which is where we perform the check that the version file is present, and seems like the obvious place to create the version file is one is not already present - as you've described. 
However, since the lotus repo and miner repo can be versioned independently (unclear if this is a requirement) this would require a `RepoType` to be passed to the `Exists()` method, which doesn't seem ideal.

Possible Solutions:
- lotus and miner repo always have the same versions: we can init the version in the Exists method as you've described
- A new `Migrate` method is created and called if exists returns `ErrNoVersion`
- Exists accepts a repotype

Reply:
I think it's ok to init the version in Exists for now